qutip/bloch.py

```diff
@@ -798,53 +784,54 @@ def plot_vectors(self):
                 self.axes.add_artist(a)
 
     def plot_points(self):
+
         # -X and Y data are switched for plotting purposes
-        for k in range(len(self.points)):
-            num = len(self.points[k][0])
-            dist = [sqrt(self.points[k][0][j] ** 2 +
-                         self.points[k][1][j] ** 2 +
-                         self.points[k][2][j] ** 2) for j in range(num)]
-            if any(abs(dist - dist[0]) / dist[0] > 1e-12):
-                # combine arrays so that they can be sorted together
-                zipped = list(zip(dist, range(num)))
-                zipped.sort()  # sort rates from lowest to highest
-                dist, indperm = zip(*zipped)
-                indperm = array(indperm)
+        for k, points in enumerate(self.points):
+            points = np.asarray(points)
+            num_points = points.shape[1]
+
+            dist = np.linalg.norm(points, axis=0)
+            if not np.allclose(dist, dist[0], rtol=1e-12):
+                indperm = np.argsort(dist)
+                points = points[:, indperm]
```

Comment:
This part of the code was also there before. I seems to sort points based on their distance to the bloch centre, but I am not sure why you would want to do so.... It is neither documented nor tested.

Reply:
I suspect this is related to the behaviour of older versions of matplotlib. Before matplotlib 3.5 the rendering order of Axes3D was determined by the zordering of the Artist objects, which is a concept that only made sense for 2D rendering. In 3.5, Axes3D gained a "computed_zorder" option that is now the default and which should order everything correctly. See https://matplotlib.org/stable/api/_as_gen/mpl_toolkits.mplot3d.axes3d.Axes3D.html.

I would vote that we remove this reordering and live with possibly suboptimal rendering in old versions of matplotlib.